### DISCUSSION TITLE
error in generating feature2

### DISCUSSION THREAD

**Discussion by Resram** [KosinskiLab/AlphaPulldown#160] [General]
After running the code snippet for feature 2, I get the following error:

(AlphaPulldown) singularity exec \
>     --no-home \
>     --bind /Ap/output:/mnt \
>      alpha-analysis.sif \
>     run_get_good_pae.sh \
>     --output_dir=/mnt \
>     --cutoff=10
I0817 11:26:01.685516 47638540354496 get_good_inter_pae.py:112] now processing P78344_feature_metadata_2023-08-11.txt
I0817 11:26:01.691167 47638540354496 get_good_inter_pae.py:112] now processing models
I0817 11:26:01.696232 47638540354496 get_good_inter_pae.py:112] now processing output.ipynb
I0817 11:26:01.701501 47638540354496 get_good_inter_pae.py:112] now processing O76094.pkl
I0817 11:26:01.706592 47638540354496 get_good_inter_pae.py:112] now processing O76094_feature_metadata_2023-08-10.txt
I0817 11:26:01.711863 47638540354496 get_good_inter_pae.py:112] now processing Q14240_feature_metadata_2023-08-11.txt
I0817 11:26:01.717097 47638540354496 get_good_inter_pae.py:112] now processing Q14240.pkl
I0817 11:26:01.722467 47638540354496 get_good_inter_pae.py:112] now processing P60842_feature_metadata_2023-08-10.txt
I0817 11:26:01.727416 47638540354496 get_good_inter_pae.py:112] now processing P60842.pkl
I0817 11:26:01.732673 47638540354496 get_good_inter_pae.py:112] now processing P78344.pkl
Traceback (most recent call last):
  File "/app/alpha-analysis/get_good_inter_pae.py", line 151, in <module>
    app.run(main)
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/app/alpha-analysis/get_good_inter_pae.py", line 141, in main
    pi_score_df=pd.merge(pi_score_df,other_measurements_df,on="jobs")
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 107, in merge
    op = _MergeOperation(
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 700, in __init__
    ) = self._get_merge_keys()
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 1110, in _get_merge_keys
    left_keys.append(left._get_label_or_level_values(lk))
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/generic.py", line 1840, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'jobs'

Please assist me to solve this issue.
Thank you in advance. 
Resram

**Comment by hqdream** [comment]
I have the same error. Any idea of how to solve this problem? Singularity is installed correctly.

**Comment by dingquanyu** [comment]
Hi @Resram ,

It seems you have mounted a wrong directory. I guess the predicted models are inside your ```models``` directory? Then the command should be:

```
--no-home \
--bind /Ap/output/models:/mnt \
 alpha-analysis.sif \
run_get_good_pae.sh \
--output_dir=/mnt \
--cutoff=10
```

**Comment by lcscs12345** [comment]
Hi @dingquanyu ,

I have a similar issue. Here is the directory structure. I tried different cutoffs and still got the same error. Any suggestions? Thanks!

```
$ tree /home/limch05p/models/
/home/limch05p/models/
└── P03452
    ├── P03452_PAE_plot_ranked_0.png
    ├── P03452_PAE_plot_ranked_1.png
    ├── P03452_PAE_plot_ranked_2.png
    ├── P03452_PAE_plot_ranked_3.png
    ├── P03452_PAE_plot_ranked_4.png
    ├── ranked_0.pdb
    ├── ranked_1.pdb
    ├── ranked_2.pdb
    ├── ranked_3.pdb
    ├── ranked_4.pdb
    ├── ranking_debug.json
    ├── result_model_1_ptm_pred_0.pkl
    ├── result_model_2_ptm_pred_0.pkl
    ├── result_model_3_ptm_pred_0.pkl
    ├── result_model_4_ptm_pred_0.pkl
    ├── result_model_5_ptm_pred_0.pkl
    ├── timings.json
    ├── unrelaxed_model_1_ptm_pred_0.pdb
    ├── unrelaxed_model_2_ptm_pred_0.pdb
    ├── unrelaxed_model_3_ptm_pred_0.pdb
    ├── unrelaxed_model_4_ptm_pred_0.pdb
    └── unrelaxed_model_5_ptm_pred_0.pdb

1 directory, 22 files
```

```
$ apptainer exec --no-home --bind /home/limch05p/models/:/mnt /home/limch05p/src/alpha-analysis_jax_0.4.sif \
  run_get_good_pae.sh \
  --output_dir=/mnt \
  --cutoff=100

I0407 08:38:39.480799 140122957333312 get_good_inter_pae.py:113] now processing P03452
I0407 08:38:39.482650 140122957333312 get_good_inter_pae.py:138] done for P03452 1 out of 1 finished.
Traceback (most recent call last):
  File "/app/alpha-analysis/get_good_inter_pae.py", line 156, in <module>
    app.run(main)
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 308, in run
    _run_main(main, args)
  File "/opt/conda/lib/python3.9/site-packages/absl/app.py", line 254, in _run_main
    sys.exit(main(argv))
  File "/app/alpha-analysis/get_good_inter_pae.py", line 146, in main
    pi_score_df=pd.merge(pi_score_df,other_measurements_df,on="jobs")
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 107, in merge
    op = _MergeOperation(
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 700, in __init__
    ) = self._get_merge_keys()
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/reshape/merge.py", line 1110, in _get_merge_keys
    left_keys.append(left._get_label_or_level_values(lk))
  File "/opt/conda/lib/python3.9/site-packages/pandas/core/generic.py", line 1840, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'jobs'
```

**Comment by dingquanyu** [comment]
Hi @lcscs12345 
It seems that your ```P03452/``` does not contains the real files but symlinks to the real files instead? From your log, it does not print the statements like 
```
finished reading result pickle for the best model.
finished obtainined the plddt values.
pi_score output for P03452 will be stored at /mnt/pi_score_outputs/P03452
...
```

so many parts of the analysis pipeline did not run in reality thereby crashing the programme.

Could you try again using the real files/

Besides, it looks like you are running analysis pipeline on monomeric models, which it will not work as many parameters, such as PAE, pDockQ are used in the cases where more than one chain exists in your PDB files. If your models are monomeric, then our analysis pipeline is not really needed.

Yours
Dingquan
